src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -107,6 +107,8 @@ public static PhysicalAddress Parse(string address)
         {
             int validCount = 0;
             bool hasDashes = false;
+            bool hasColons = false;
+            bool hasPeriods = false;
```

Comment:
Rather than having these three bools, how about instead having a `maxValidCount`?  Instead of setting one of these bools to true, the relevant code would set `maxValidCount` to the right value, and then instead of having the multiple if checks later that throw, there could be just one that compared `validCount` to `maxValidCount`